packages/react-form/src/hooks/test/form.test.tsx

```diff
@@ -216,6 +216,27 @@ describe('useForm', () => {
       });
     });
 
+    it('validates list fields with their latest values before submitting and bails out if any fail', async () => {
```

Comment:
I've made a brand new test to make sure that the `submitSpy` does not get called only because of a list error.